### ISSUE TITLE
[Masthead A11Y]: Content is not within a landmark element

### ISSUE THREAD

**Issue by mo-alaa** [carbon-design-system/carbon-for-ibm-dotcom#9679]
### Description

Hi, while using the L1 masthead (contained in DotcomShell @ ibmdotcom-react), we can find that the lower navigation buttons are not  included within a landmark element.
This was raised as a **violation** while scanning [the page](https://www.ibm.com/impact) (L1 is used in that page) with [IBM Equal Access Accessibility Checker](https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp/related).

I also think that all navigation buttons in masthead of _ibmdotcom-web-components_ are not included in a landmark element.
Thanks!
![image (15)](https://user-images.githubusercontent.com/41386480/203320729-648b09b0-9469-48ed-8594-37ab9a3a8681.png)

### Component(s) impacted

DotcomShell of ibmdotcom-react

### Browser

Chrome

### Carbon for IBM.com version

@carbon/ibmdotcom-react: "^1.31.1"

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

ibm.com/impact

### Package

@carbon/ibmdotcom-react, @carbon/ibmdotcom-web-components

### CodeSandbox example

https://www.ibm.com/standards/carbon/react/?path=/story/components-masthead--with-platform

### Steps to reproduce the issue (if applicable)

This can be inspected in the [docs for the masthead L1 element](https://www.ibm.com/standards/carbon/react/?path=/story/components-masthead--with-platform) as shown in the image.
![image (16)](https://user-images.githubusercontent.com/41386480/203320847-1bfdd9cd-8d43-4269-ab95-c9bd57780dd5.png)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by mo-alaa** [comment]
Just a heads-up. Thanks!

**Comment by sangeethababu9223** [comment]
@mo-alaa ,

The lower navigation buttons are inside element with "main" role.
IBM Equal Access Accessibility Checker is not listing the issue either.
